src/Shared-Libs/ManagedLoader/Datadog.AutoInstrumentation.ManagedLoader/public/AssemblyLoader.cs

```diff
@@ -76,45 +77,55 @@ public AssemblyLoader(string[] assemblyNamesToLoadIntoDefaultAppDomain, string[]
         /// <param name="assemblyNamesToLoadIntoNonDefaultAppDomains">List of assemblies to load and start if the curret App Domain is the NOT default App Domain.</param>
         public static void Run(string[] assemblyNamesToLoadIntoDefaultAppDomain, string[] assemblyNamesToLoadIntoNonDefaultAppDomains)
         {
-            try
-            {
+            /*
+             * Here we delay the loading of the assemblies to prevent a crash due to some unknown (yet) race.
+             * The case has been seen with a WCF application (client: Inovalon)

```

Comment:
This is an open-source repository. Let's leave customer names out of the source code.
```suggestion
             * The case has been seen with a WCF application.

```

Reply:
Thanks for your review @lucaspimentel